### PR TITLE
Remove monix

### PR DESCRIPTION
**Pull request by danilbykov**
Changes:

1. Remove monix tasks and monix observables.
Class `monix.reactive.Observable` was core of `Reloadable`s. Now it is removed and module core depends only on cats (without cats effects or zio). Instead of  `monix.reactive.Observable` new class `RawReloadable` is implemented with similar interface. The main method `RawReloadable.subscribe` allows to listen changes in this reloadable.
Based on `RawReloadable` from module core implemented two modules core-ce and core-zio. These modules define `Reloadable` and `ReactiveConfig`. Interfaces of these classes in core-ce are almost not changed.

2. Better interface for `zio.Reloadable`.
`Reloadable` from core-ce has only type parameter `F[_]` which is usually defined by `ReactiveConfig` so it is often some effect which may fail (`zio.IO[E, A]`). But result of `Reloadable.get` is also wrapped in `F[_]` which means it may fail. But it is impossible by implementation because it always returns `zio.UIO`. So for core-zio `zio.Reloadable` has slightly different interface which takes advantage of zio:
- `zio.Reloadable.get` always returns result in `zio.UIO` which simplifies usage in user's code and doesn't require additional error handling
- `zio.Reloadable.map` returns `UManaged` which can not fail
- similar improvements for another methods

3. Reloadable.combine to combine several Reloadables into one.
`Reloadable` allows to combine only two `Reloadable`s at once (using method `Reloadable.combine`) which is not convenient when several reloadables must be combined. So added boilerplate code which combines up to 22 reloadables at once. This simplifies user's code and also has better performance because these methods create only one reloadable (without intermediate reloadables).

4. Annotations `@reactiveconfig` and `@source` for Reloadable derivation.
These annotations allow build `Reloadable`s for case classes when each field is mapped by some path in config (etcd or typesafe config).

### CODE REVIEW

File: core/src/main/scala/com/github/fit51/reactiveconfig/reloadable/RawReloadable.scala [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    var current = ref.get()
+    while (!ref.compareAndSet(current, mod(current)))
+      current = ref.get()
+    current
+  }
+
+  @annotation.tailrec
+  def getAndUpdate[X, A](ref: AtomicReference[X])(mod: X => (X, A)): A = {
+    var current        = ref.get()
+    var (newX, result) = mod(current)
```
Comment (realvikr) [inline core/src/main/scala/com/github/fit51/reactiveconfig/reloadable/RawReloadable.scala]: val

File: core-zio/src/main/scala/com/github/fit51/reactiveconfig/zio/reloadable/Reloadable.scala [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    new ConstReloadable(a)
+
+  implicit class ResourceLikeOps[R, E, A](private val resource: ResourceLike[ZIO[R, E, *], A]) extends AnyVal {
+    def toManaged: ZManaged[R, E, A] = {
+      def go[B](resource: ResourceLike[ZIO[R, E, *], B]): ZManaged[R, E, B] =
+        resource match {
+          case Allocate(resource) =>
+            // FIXME: uncacellable
+            Managed.makeReserve(resource.map { case (b, release) =>
+              Reservation(ZIO.succeed(b), _ => release.catchAll(_ => UIO.unit))
```
Comment (realvikr) [inline core-zio/src/main/scala/com/github/fit51/reactiveconfig/zio/reloadable/Reloadable.scala]: `release.ignore`

File: core/src/main/scala/com/github/fit51/reactiveconfig/reloadable/subscribers.scala [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    state.fold(Effect[F].unit)(f(_).flatMap(_.fold(Effect[F].unit)(foreach)))
+
+  override protected def updateState(state: Option[AA], update: AA): Option[AA] =
+    Some(update)
+
+  override def onNext(next: AA): F[Unit] =
+    modifyState(next)
+}
+
+private class CombinedFSubsriber[F[_]: Effect, A, B, C](
```
Comment (realvikr) [inline core/src/main/scala/com/github/fit51/reactiveconfig/reloadable/subscribers.scala]: CombinedFSubs**c**riber

File: typesafe/src/main/scala/com/github/fit51/reactiveconfig/typesafe/TypesafeUtils.scala [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+import java.nio.file.Path
+import java.util
+
+import com.github.fit51.reactiveconfig.Value
+import com.github.fit51.reactiveconfig.parser.ConfigParser
+import com.github.fit51.reactiveconfig.typeclasses.Effect
+import com.typesafe.config._
+
+import scala.collection.JavaConverters._
```
Comment (realvikr) [inline typesafe/src/main/scala/com/github/fit51/reactiveconfig/typesafe/TypesafeUtils.scala:11]: use `import scala.jdk.CollectionConverters._`

File: typesafe/src/main/scala/com/github/fit51/reactiveconfig/typesafe/TypesafeUtils.scala [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            parseHoconEntryToJson(subEntry),
+            storage
+          )
+        }
+      case _ => updatedStorage
+    }
+  }
+
+  def parseValuesInMap[F[_], D](map: Map[String, Try[D]], index: Long)(implicit
+      encoder: ConfigParser[D],
```
Comment (realvikr) [inline typesafe/src/main/scala/com/github/fit51/reactiveconfig/typesafe/TypesafeUtils.scala:63]: not used

File: core-zio/src/main/scala/com/github/fit51/reactiveconfig/zio/reloadable/Reloadable.scala (hunk 2 above)
Comment (danilbykov) [reply core-zio/src/main/scala/com/github/fit51/reactiveconfig/zio/reloadable/Reloadable.scala]: Actually we have `logger` here (which comes from `EffectInstances`) so I decided to print these errors in log
```scala
UIO.effectTotal(logger.error(s"Unable to release: $e"))
```

File: core/src/main/scala/com/github/fit51/reactiveconfig/reloadable/RawReloadable.scala (hunk 1 above)
Comment (danilbykov) [reply core/src/main/scala/com/github/fit51/reactiveconfig/reloadable/RawReloadable.scala]: Fixed

File: core/src/main/scala/com/github/fit51/reactiveconfig/reloadable/subscribers.scala (hunk 3 above)
Comment (danilbykov) [reply core/src/main/scala/com/github/fit51/reactiveconfig/reloadable/subscribers.scala]: Fixed

File: typesafe/src/main/scala/com/github/fit51/reactiveconfig/typesafe/TypesafeUtils.scala (hunk 4 above)
Comment (danilbykov) [reply typesafe/src/main/scala/com/github/fit51/reactiveconfig/typesafe/TypesafeUtils.scala:11]: It breaks compilation of scala2.12 modules.

File: typesafe/src/main/scala/com/github/fit51/reactiveconfig/typesafe/TypesafeUtils.scala (hunk 5 above)
Comment (danilbykov) [reply typesafe/src/main/scala/com/github/fit51/reactiveconfig/typesafe/TypesafeUtils.scala:63]: No, it is used from classes `TypesafeReactiveConfig` from modules typesafe-zio and typesafe-ce.

**Review by realvikr — APPROVED** [review]

**Pull request merged**